docs: Add checkpoints, overview, and Redis to Seqera AI install guide (#1266)
seqeralabs/docs: platform-enterprise_docs/enterprise/install-seqera-ai.md

Seqera AI is an intelligent command-line assistant that helps you build, run, and manage bioinformatics workflows. This guide describes how to deploy Seqera AI in a Seqera Enterprise deployment.

## Overview

> Seqera AI enables users to interact with Seqera Platform through a conversational AI interface, available through both the web (portal) and the CLI. Deploying Seqera AI involves standing up the following components in sequence:

Reviewer: @odagayev - what happens if the sequence isn't followed? Can the user expect an error? Is there a way to remediate?

Author: nothing really - just nothing will work until you don't set everything up... hahaha

| Step | Component | Purpose |
|------|-----------|---------|
| 1 | **MCP server** | Model Context Protocol server that provides Platform-aware tools (workflows, datasets, compute environments). Deploy this first. |
| 2 | **MySQL database** | Stores session state and conversation history. Can share the same database instance as Platform or use a dedicated instance. |
| 3 | **Redis** | Caching and session management layer for the agent backend. |
| 4 | **Agent backend** | FastAPI service that orchestrates AI interactions between the CLI/web, the inference provider, and MCP. Deployed as a Helm subchart alongside Platform. |
| 5 | **Portal web interface** | Browser-based interface for Seqera AI and related Platform features. |

Each step below includes a verification checkpoint so you can confirm the component is working before moving to the next.

## Prerequisites

Before you begin, you need:

> - **Seqera Enterprise 26.1+** deployed and accessible. Platform can be deployed via [Helm](./platform-helm.md), [Kubernetes](./platform-kubernetes.md), or [Docker Compose](./platform-docker-compose.md) — a Helm-based deployment is not required.

Reviewer 2: ```suggestion
- **Seqera Enterprise 26.1+** deployed and accessible.
```
At  most we should link back to the front page for the platform install docs, we do not have to rehash or make comment that Helm-Based install is not required.

> - **MySQL 8.0+ database**

Reviewer 3: Shouldn't kubernetes be a requirement too?

> - **Redis 6.0+** instance accessible from your cluster

Reviewer 2: This does not align to the other language used in the installation documentation. 

Redis 6 is no longer supported and EOL and not compatible with other platform components.

    ```
- [Helm v3](https://helm.sh/docs/intro/install) and [kubectl](https://kubernetes.io/docs/tasks/tools/) installed locally

### Permissions

The person performing this installation needs:

- **Kubernetes cluster access**: `kubectl` access with permissions to create and manage deployments, services, secrets, and ingress resources in the target namespace
- **Database administration**: Ability to create databases and users on the MySQL instance
- **DNS management**: Access to create or update DNS records for the Seqera AI subdomains
- **AWS IAM** (if using Bedrock): Permissions to invoke Bedrock models in the target AWS account

> - **Secret management**: Access to create Kubernetes secrets in the target namespace

Reviewer 2: Secrets management is a deployment choice and not a requirement.

- **Helm**: Permission to install and upgrade Helm releases in the target namespace
- **TLS certificate management**: Access to provision or reference TLS certificates (e.g., via AWS Certificate Manager)

> 

Reviewer 2: This should move up to requirements.

| **Agent backend** | FastAPI service that orchestrates AI interactions. Deployed as a Helm subchart alongside Platform. |
| **MCP server** | Model Context Protocol server providing Platform-aware tools (workflows, datasets, compute environments). |
| **Portal web interface** | Browser-based interface for Seqera AI and related Platform features. |

> | **MySQL database** | Database for session state and conversation history. Can share the same instance as Platform (recommended) or use a dedicated instance. |

Reviewer 2: ```suggestion
| MySQL database | Dedicated database for session state and conversation history. 
```

From claudio

The original wording correctly states the requirement for a dedicated database separate from the Platform database. The proposed change introduces deployment methodology detail which is out of scope for a requirements. 

Database = A logically separated, named collection of structured data with its own schema, access controls, and namespace.
Instance = A running deployment of a database engine that may host one or more databases.

1. MCP tools execute Platform operations using the user's credentials.
1. Results stream back to the CLI via Server-Sent Events (SSE).

> ## Step 1: Deploy the MCP server

Reviewer 2: We should move this to it's own stand alone install guide. ( Noted for future improvement)

Deploy the MCP server using the [MCP Helm chart](https://github.com/seqeralabs/helm-charts/tree/master/platform/charts/mcp). The MCP server can be installed alongside Platform in a single Helm release or as a separate release.

### Checkpoint: Verify MCP is running

Confirm the MCP server is healthy and reachable from within your cluster:

```bash
curl -i https://mcp.platform.example.com/health
curl -i https://mcp.platform.example.com/service-info
```

You should receive `200 OK` responses. If these fail, resolve MCP connectivity before proceeding.

:::tip
If you see connection errors, verify:
- The MCP server pod is running (`kubectl get pods` in the MCP namespace)
- Network policies or security groups allow traffic to the MCP endpoint
- DNS resolves correctly for the MCP domain
:::

## Step 2: Provision the database

Seqera AI requires its own MySQL database for session state and conversation history. You can create this database on the same instance that hosts your Platform database (recommended) or on a separate dedicated instance.

:::tip Recommended: Use the same database instance as Platform

> Creating the Seqera AI database on your existing Platform database instance simplifies infrastructure management, reduces costs, and avoids additional networking configuration. The Seqera AI database is lightweight and does not compete for resources with the Platform database under typical usage.

Reviewer: Do we have an idea of "typical usage" parameters? If there are competing resource requirements where can this be managed?

Reviewer 2: This should not be a recommendation. 

We should not be advising customers on how they do their database techologies.

Seqera AI should be on it's own DB schema separate from platform, it can live on it's own host.

Note recommending infrastructure goes against the persona the installation documentation is intended for which is an experienced system administrator.

Reviewer: @odagayev - where in the Enterprise docs would this content live? If it's meant as an installation guide, it would go under this section: https://docs.seqera.io/platform-enterprise/enterprise/overview and should follow the same structure and persona. 

@gavinelder - I know we have a persona file for this; is there an opportunity to create a template for this/other content that follows the current structure and content guidelines?

Author: @gwright99 - thoughts here? what's the recommendation from your experience?

If you have strict isolation requirements, you can provision a separate instance instead.
:::

Connect to your database instance and create the Seqera AI database and user:

```sql
CREATE DATABASE seqera_ai;
CREATE USER 'seqera_ai'@'%' IDENTIFIED BY '<secure-password>';
GRANT ALL PRIVILEGES ON seqera_ai.* TO 'seqera_ai'@'%';
FLUSH PRIVILEGES;
```

> 

Reviewer 3: ```suggestion
```
do we need to provide instructions on how to set up a mysql db? are these instructions generic to all mysql implementations?

> ### Checkpoint: Verify database connectivity
> 
> Confirm the database is accessible from your cluster:
> 
> ```bash
> kubectl run db-check --rm -it --restart=Never \
>     --image=mysql:8.0 -- \
>     mysql -h <db-hostname> -u seqera_ai -p<secure-password> -e "SELECT 1;"
> ```
> 
> You should see a result set with the value `1`. If this fails, check security group rules and network connectivity.
> 

Reviewer 3: ```suggestion
```

This feels like out of scope for this guide
The helm chart already has an init container that stops until the db is available

> ## Step 3: Provision Redis

Reviewer 2: This is a requirement and not something we specify in documentation. 

The target persona should know how to do this.

> Seqera AI requires a Redis instance for caching and session management. Use a managed Redis service or a self-managed instance, as long as it is accessible from your cluster on port 6379.

Reviewer 3: ```suggestion
Seqera AI requires a Redis-compatible instance for caching and session management. Seqera recommends a managed Redis service.
```

> - **Engine**: Redis 6.0+

Reviewer 3: were the apps tested with redis 6?

> - **Security group**: Allow inbound Redis (port 6379) from your cluster

Reviewer 3: ```suggestion
```

this is out of scope for this guide

> - **Security group**: Allow inbound Redis (port 6379) from your cluster

Reviewer 2: Redundant and cloud specific.
